test/integration/main_test.go

```diff
@@ -39,6 +40,33 @@ func KubeClient() (result *kubernetes.Clientset) {
 	return kubeClient
 }
 
+func COInsights(k *kubernetes.Clientset) map[string]interface{} {
+	// get info about insights cluster operator
+	data, err := k.RESTClient().Get().AbsPath("/apis/config.openshift.io/v1/clusteroperators/insights").DoRaw()
+	obj := map[string]interface{}{}
+	err = json.Unmarshal(data, &obj)
+	if err != nil {
+		panic(err.Error())
+	}
+	return obj
+}
+
+func IsOperatorDegraded(t *testing.T, config map[string]interface{}) bool {
+	status := config["status"].(map[string]interface{})
```

Comment:
Could we use the plumbing around cluster operator API to deserialize the data, so that we don't need to deal with the `map[string]interface{}` which is pretty painful to work with.

I guess code in https://github.com/openshift/insights-operator/blob/fac81f249a23886f976a8869e38978b2a2355275/pkg/controller/status/status.go#L346 might be a good example on how to use it.

Reply:
thank you! I found a way to remove this ugly code :)